controllers/dspipeline_controller.go

```diff
@@ -197,57 +200,203 @@ func (r *DSPAReconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.
 		return ctrl.Result{}, nil
 	}
 
-	usingCustomDB := params.UsingExternalDB(dspa)
+	// Initialize conditions
+	conditions := []metav1.Condition{}
 
+	usingCustomDB := params.UsingExternalDB(dspa)
+	databaseReady := r.buildCondition("DatabaseReady", dspa.Generation, "DatabaseReady")
 	if !usingCustomDB {
+
 		err = r.ReconcileDatabase(ctx, dspa, params)
 		if err != nil {
+			databaseReady.Status = metav1.ConditionFalse
+			databaseReady.Reason = "DataBaseFailedToDeploy"
```

Comment:
can we store all `.Reason` as constants in one place as variables e.g.: https://github.com/opendatahub-io/data-science-pipelines-operator/blob/main/controllers/config/defaults.go